plugin/pkg/scheduler/algorithm/priorities/selector_spreading_test.go

```diff
@@ -228,6 +229,203 @@ func TestSelectorSpreadPriority(t *testing.T) {
 	}
 }
 
+func TestZoneSelectorSpreadPriority(t *testing.T) {
+	labels1 := map[string]string{
+		"label1": "l1",
+		"baz":    "blah",
+	}
+	labels2 := map[string]string{
+		"label2": "l2",
+		"baz":    "blah",
+	}
+	nodeLabelsZone1 := map[string]string{
+		wellknownlabels.LabelZoneFailureDomain: "zone1",
+	}
+	nodeLabelsZone2 := map[string]string{
+		wellknownlabels.LabelZoneFailureDomain: "zone2",
+	}
+	nodeLabelsZone3 := map[string]string{
+		wellknownlabels.LabelZoneFailureDomain: "zone3",
+	}
+	labeledNodes := map[string]map[string]string{
+		"machine1.zone1": nodeLabelsZone1,
```

Comment:
We definitely have unequal number of nodes per zone: zone1 has 1 node, zone2 has 2 nodes, zone3 has 3 nodes.

I replaced the nodeLabels with a generator function. I also replaced the string constants with constant variables, so the compiler will pick up any typos.  I'm happy to do anything better but I can't see what that would be - the problem is that some of the later test cases have some pretty complicated behavior.